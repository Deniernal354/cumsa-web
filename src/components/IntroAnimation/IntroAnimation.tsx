"use client";
import { useEffect, useState } from "react";
import Image from "next/image";

export default function IntroAnimation() {
  const [show, setShow] = useState(true);
  const [phase, setPhase] = useState<"show" | "disperse" | "exit">("show");
  const [hasPlayed, setHasPlayed] = useState(false);

  useEffect(() => {
    // Check if intro already played
    const introPlayed = sessionStorage.getItem("cumsa-intro-played");

    if (introPlayed) {
      setShow(false);
      setHasPlayed(true);
      return;
    }

    // Mark as played immediately
    sessionStorage.setItem("cumsa-intro-played", "true");

    // Simple timeline - show for 2 seconds, then disperse
    const disperseTimer = setTimeout(() => setPhase("disperse"), 2000);
    const exitTimer = setTimeout(() => {
      setPhase("exit");
      setTimeout(() => {
        setShow(false);
        setHasPlayed(true);
      }, 400); // Reduced from 800ms to 400ms
    }, 2500); // Reduced from 2800ms to 2500ms

    return () => {
      clearTimeout(disperseTimer);
      clearTimeout(exitTimer);
    };
  }, []);

  // Don't render if already played
  if (hasPlayed || !show) return null;

  return (
    <div
      className={`fixed inset-0 z-50 transition-all duration-400 ease-in-out
        ${phase === "exit" ? "opacity-0" : "opacity-100"}
      `}
      style={{
        background: `linear-gradient(135deg, 
          rgba(90, 206, 216, 0.1) 0%, 
          rgba(255, 255, 255, 0.95) 40%, 
          rgba(255, 255, 255, 0.95) 60%, 
          rgba(42, 116, 135, 0.1) 100%)`,
      }}
    >
      {/* Floating particles background */}
      <div className="absolute inset-0 overflow-hidden pointer-events-none">
        {[...Array(6)].map((_, i) => (
          <div
            key={i}
            className={`absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              ${phase === "show" ? "animate-pulse" : ""}
              ${phase === "disperse" ? "animate-ping" : ""}
            `}
            style={{
              left: `${15 + i * 15}%`,
              top: `${20 + (i % 2) * 60}%`,
              animationDelay: `${i * 0.5}s`,
              animationDuration: "3s",
            }}
          />
        ))}
      </div>

      {/* Main content - perfectly centered */}
      <div className="absolute inset-0 flex flex-col items-center justify-center px-4 sm:px-8">
        {/* Logo container with animations */}
        <div className="relative flex flex-col items-center">
          {/* Logo with disperse effects - NO DELAYS */}
          <div
            className={`relative transition-all duration-700 ease-out mb-6 sm:mb-8
              ${phase === "show" ? "scale-100 opacity-100 translate-y-0" : ""}
              ${phase === "disperse" ? "scale-150 opacity-0" : ""}
              ${phase === "exit" ? "scale-125 opacity-0" : ""}
            `}
            style={{
              filter: phase === "disperse" ? "blur(2px)" : "none",
            }}
          >
            {/* Glow effect */}
            <div
              className={`absolute inset-0 rounded-full transition-all duration-1000
                ${phase === "show" ? "bg-[var(--blue)] opacity-20 scale-110 blur-xl" : "opacity-0 scale-100"}
                ${phase === "disperse" ? "scale-200 opacity-0" : ""}
              `}
            />

            {/* Main logo */}
            <Image
              src="/CU-MSA-LOGO.png"
              alt="CUMSA Logo"
              width={160}
              height={160}
              className={`relative z-10 
                ${phase === "show" ? "animate-bounce" : ""}
                sm:w-48 sm:h-48
              `}
              style={{
                animationDuration: phase === "show" ? "2s" : "0s",
                animationIterationCount: "infinite",
              }}
              priority
            />

            {/* Disperse particles */}
            {phase === "disperse" && (
              <div className="absolute inset-0">
                {[...Array(8)].map((_, i) => (
                  <div
                    key={i}
                    className="absolute w-3 h-3 bg-[var(--blue)] rounded-full animate-ping"
                    style={{
                      left: `${50 + Math.cos((i * Math.PI) / 4) * 100}%`,
                      top: `${50 + Math.sin((i * Math.PI) / 4) * 100}%`,
                      animationDelay: `${i * 0.1}s`,
                      animationDuration: "0.8s",
                    }}
                  />
                ))}
              </div>
            )}
          </div>

          {/* Brand text - NO DELAYS, INSTANT APPEARANCE */}
          <div
            className={`text-center transition-all duration-700 ease-out
              ${phase === "show" ? "opacity-100 translate-y-0" : ""}
              ${phase === "disperse" ? "opacity-70 scale-105" : ""}
              ${phase === "exit" ? "opacity-0 translate-y-2" : ""}
            `}
          >
            <h1 className="text-2xl sm:text-4xl font-bold text-[var(--navy)] mb-2 sm:mb-3 tracking-wide">
              CUMSA
            </h1>
            <p className="text-sm sm:text-base text-slate-600 font-light px-4 leading-relaxed">
              Carleton University Muslim Students&apos; Association
            </p>

            {/* Decorative line - INSTANT */}
            <div
              className={`w-16 sm:w-24 h-0.5 bg-gradient-to-r from-[var(--navy)] to-[var(--blue)] 
                mx-auto mt-3 sm:mt-4 transition-all duration-500
                ${phase === "show" ? "opacity-60 scale-100" : "opacity-0 scale-75"}
              `}
            />
          </div>

          {/* Welcome text - INSTANT */}
          <p
            className={`text-xs sm:text-sm text-slate-500 mt-4 sm:mt-6 transition-all duration-500
              ${phase === "show" ? "opacity-100 translate-y-0" : "opacity-0 translate-y-2"}
              ${phase === "disperse" ? "opacity-30" : ""}
            `}
          >
            ✨ Welcome ✨
          </p>
        </div>
      </div>

      {/* Bottom gradient fade */}
      <div className="absolute bottom-0 left-0 right-0 h-32 bg-gradient-to-t from-white/50 to-transparent pointer-events-none" />
    </div>
  );
}
