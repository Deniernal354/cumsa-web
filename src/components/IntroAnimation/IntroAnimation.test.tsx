import { render } from "@testing-library/react";
import IntroAnimation from "./IntroAnimation";

test("IntroAnimation snapshot", () => {
  const { asFragment } = render(<IntroAnimation />);
  expect(asFragment()).toMatchSnapshot();
});
