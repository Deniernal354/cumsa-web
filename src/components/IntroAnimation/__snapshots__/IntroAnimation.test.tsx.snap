// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroAnimation snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 z-50 transition-all duration-400 ease-in-out
        opacity-100
      "
    style="background: linear-gradient(135deg, rgba(90, 206, 216, 0.1) 0%, rgba(255, 255, 255, 0.95) 40%, rgba(255, 255, 255, 0.95) 60%, rgba(42, 116, 135, 0.1) 100%);"
  >
    <div
      class="absolute inset-0 overflow-hidden pointer-events-none"
    >
      <div
        class="absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              animate-pulse
              
            "
        style="left: 15%; top: 20%; animation-delay: 0s; animation-duration: 3s;"
      />
      <div
        class="absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              animate-pulse
              
            "
        style="left: 30%; top: 80%; animation-delay: 0.5s; animation-duration: 3s;"
      />
      <div
        class="absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              animate-pulse
              
            "
        style="left: 45%; top: 20%; animation-delay: 1s; animation-duration: 3s;"
      />
      <div
        class="absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              animate-pulse
              
            "
        style="left: 60%; top: 80%; animation-delay: 1.5s; animation-duration: 3s;"
      />
      <div
        class="absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              animate-pulse
              
            "
        style="left: 75%; top: 20%; animation-delay: 2s; animation-duration: 3s;"
      />
      <div
        class="absolute w-2 h-2 bg-[var(--blue)] rounded-full opacity-20
              animate-pulse
              
            "
        style="left: 90%; top: 80%; animation-delay: 2.5s; animation-duration: 3s;"
      />
    </div>
    <div
      class="absolute inset-0 flex flex-col items-center justify-center px-4 sm:px-8"
    >
      <div
        class="relative flex flex-col items-center"
      >
        <div
          class="relative transition-all duration-700 ease-out mb-6 sm:mb-8
              scale-100 opacity-100 translate-y-0
              
              
            "
          style="filter: none;"
        >
          <div
            class="absolute inset-0 rounded-full transition-all duration-1000
                bg-[var(--blue)] opacity-20 scale-110 blur-xl
                
              "
          />
          <img
            alt="CUMSA Logo"
            class="relative z-10 
                animate-bounce
                sm:w-48 sm:h-48
              "
            height="160"
            src="/CU-MSA-LOGO.png"
            style="animation-duration: 2s; animation-iteration-count: infinite;"
            width="160"
          />
        </div>
        <div
          class="text-center transition-all duration-700 ease-out
              opacity-100 translate-y-0
              
              
            "
        >
          <h1
            class="text-2xl sm:text-4xl font-bold text-[var(--navy)] mb-2 sm:mb-3 tracking-wide"
          >
            CUMSA
          </h1>
          <p
            class="text-sm sm:text-base text-slate-600 font-light px-4 leading-relaxed"
          >
            Carleton University Muslim Students' Association
          </p>
          <div
            class="w-16 sm:w-24 h-0.5 bg-gradient-to-r from-[var(--navy)] to-[var(--blue)] 
                mx-auto mt-3 sm:mt-4 transition-all duration-500
                opacity-60 scale-100
              "
          />
        </div>
        <p
          class="text-xs sm:text-sm text-slate-500 mt-4 sm:mt-6 transition-all duration-500
              opacity-100 translate-y-0
              
            "
        >
          ✨ Welcome ✨
        </p>
      </div>
    </div>
    <div
      class="absolute bottom-0 left-0 right-0 h-32 bg-gradient-to-t from-white/50 to-transparent pointer-events-none"
    />
  </div>
</DocumentFragment>
`;
